### PR TITLE
Fixes an  issue, when updateObject is used for updating a row in a table based on an object's properties..

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes

Fixes an  issue, when updateObject is used for updating a row in a table based on an object's properties..

Specifically the problem is when null is used in the `WHERE` clause (not to be confused with `NULL`'s being updated).

### Testing Instructions

Not sure if any needed. It should not change behavior (except not being a wrong query ;) ).
Try this:
Create a table:
```
CREATE TABLE test
(
  id INT(11) PRIMARY KEY NOT NULL,
  value VARCHAR(255)
);
INSERT INTO `test` (`id`, `value`) VALUES (NULL, NULL), (NULL, 'test');
```

```
SELECT * FROM `test` WHERE value = null;
```
Notice that the correct row is not returned;

```
SELECT * FROM `test` WHERE value is null;
```
Notice that the row with the NULL value is returned.

Also before this specific change, the quote function would essentially write the query as such: 
```
SELECT * FROM `test` WHERE value = '';
```
... which is also wrong

### Documentation Changes Required
None.

### CODE REVIEW

**Review by mbabker — APPROVED** [review]

File: libraries/joomla/database/driver.php [hunk 1]
```diff
@@ -2220,15 +2220,15 @@ public function updateObject($table, &$object, $key, $nulls = false)
 		foreach (get_object_vars($object) as $k => $v)
 		{
 			// Only process scalars that are not internal fields.
-			if (is_array($v) or is_object($v) or $k[0] == '_')
+			if (is_array($v) or is_object($v) or $k[0] === '_')
```
Comment (Quy) [inline libraries/joomla/database/driver.php]: Change `or` to `||`.

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 281ac00813249d99a5768b1459e34743616c729d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13733">issues.joomla.org/tracker/joomla-cms/13733</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 281ac00813249d99a5768b1459e34743616c729d<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13733">issues.joomla.org/tracker/joomla-cms/13733</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**